app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,14 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @posts = Post.includes(:author)
+      .includes(:comments)
+      .where(author: params[:user_id])
+      .order(created_at: :asc)
 
-  def show; end
+    @author = @posts.first.author
```

Comment:
**OPTIONAL** - You have done well so far, but to help with the readability and efficiency of your code, I suggest your index controller is better written as:
```ruby 
     def index
         @author = User.find(params[:user_id])
         @posts = @author.posts.includes(:comments)
     end
```
This way you only need to search for the user and get his posts, instead of searching through all posts with the author_id of the user. Please fix this 🙏

Reply:
Thanks for giving the suggestion for these changes, I will make these changes to improve the readability and efficiency of the code.